site/src/pages/RoadmapPage/transfers/GESection.tsx

```diff
@@ -1,8 +1,154 @@
-import { FC } from 'react';
-import MenuSection from './MenuSection';
+import './GESection.scss';
+import { FC, useState, useEffect } from 'react';
+import MenuSection, { SectionDescription } from './MenuSection';
+import MenuTile from './MenuTile';
+import { GEName, GETitle, TransferredGE } from '@peterportal/types';
+import trpc from '../../../trpc';
+import { setAllTransferredGEs, setTransferredGE } from '../../../store/slices/transferCreditsSlice';
+import { useAppDispatch, useAppSelector } from '../../../store/hooks';
+
+const ALL_GE_NAMES: GEName[] = ['GE-1A', 'GE-1B', 'GE-2', 'GE-3', 'GE-4', 'GE-5A', 'GE-5B', 'GE-6', 'GE-7', 'GE-8'];
+const GE_TITLE_MAP: Record<GEName, GETitle> = {
+  'GE-1A': 'GE Ia: Lower Division Writing',
+  'GE-1B': 'GE Ib: Upper Division Writing',
+  'GE-2': 'GE II: Science and Technology',
+  'GE-3': 'GE III: Social & Behavioral Sciences',
+  'GE-4': 'GE IV: Arts and Humanities',
+  'GE-5A': 'GE Va: Quantitative Literacy',
+  'GE-5B': 'GE Vb: Formal Reasoning',
+  'GE-6': 'GE VI: Language Other Than English',
+  'GE-7': 'GE VII: Multicultural Studies',
+  'GE-8': 'GE VIII: International/Global Issues',
+};
+
+interface GEInputProps {
+  value: number;
+  handleUpdate: (newValue: number) => void;
+  valueType: 'numberOfCourses' | 'units';
+}
+
+const GEInput: FC<GEInputProps> = ({ value, handleUpdate, valueType }) => {
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const allowedKeys = [
+      '0',
+      '1',
+      '2',
+      '3',
+      '4',
+      '5',
+      '6',
+      '7',
+      '8',
+      '9',
+      valueType === 'units' ? '.' : '',
+      'Backspace',
+      'Tab',
+      'ArrowLeft',
+      'ArrowRight',
+    ];
```

Comment:
I just realized this is `onKeyDown` instead of `onInput`. I highly recommend you change it to onInput, because then you can control the _value_ change rather than directly modify behavior of the key being pressed.

This is relevant in cases such as pressing the up/down arrows to increment/decrement the value and using ctrl + A to select all. I know I did suggest a whitelist, but if that's the case it should be with `oninput`, or just replace all invalid characters `const invalidCharExp = valueType === 'units' ? /[^0-9.]/g : /\D/g` with empty string

Another alternative could be to allow invalid inputs to show up, but only call `handleUpdate` if Number(e.target.value) and e.target.valid. I can try to take a look at that though since it seems to be a minor blocker

Reply:
updated your branch, would like to merge so we can count units in roadmap. If there are issues, please lmk separately and we will address them before final merge of redesigned transfers